security-analysis/security-analysis-api/src/main/java/com/powsybl/security/interceptors/SecurityAnalysisInterceptor.java

```diff
@@ -15,9 +16,44 @@
  */
 public interface SecurityAnalysisInterceptor {
 
+    /**
+     * Callback after the pre-contingency analysis result is created
+     * @param context The running context
+     * @param preContingencyResult
+     */
     void onPreContingencyResult(RunningContext context, LimitViolationsResult preContingencyResult);
 
+    /**
+     * Deprecated. Use {@link #onPostContingencyResult(ContingencyContext context, PostContingencyResult postContingencyResult)}
+     * @param context
+     * @param postContingencyResult
+     */
+    @Deprecated
     void onPostContingencyResult(RunningContext context, PostContingencyResult postContingencyResult);
```

Comment:
Should we provide a default implementation so that new implementations do not have to implement this deprecated method ?

Reply:
The new InterceptorImpl should extends `DefaultSecurityAnalysisInterceptor`

Reply:
You cannot assume that: a developper could inherit from the interface directly. If you deprecate a method, you have to provide a default implementation to make it possible to remove it without breaking things.
Without a default implementation, I need to implement it in my subclass so removing this method from the API will break my code.